src/main/scala/io/scalac/tezos/translator/service/LibraryService.scala

```diff
@@ -1,45 +1,53 @@
 package io.scalac.tezos.translator.service
 
+import akka.event.LoggingAdapter
 import io.scalac.tezos.translator.model.LibraryEntry._
 import io.scalac.tezos.translator.model.{LibraryEntry, Uid}
 import io.scalac.tezos.translator.repository.LibraryRepository
 import io.scalac.tezos.translator.repository.dto.LibraryEntryDbDto
-import slick.jdbc.PostgresProfile.api._
-import io.scalac.tezos.translator.service.LibraryService.UidNotExists
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
 
-class LibraryService(repository: LibraryRepository, db: Database)(implicit ec: ExecutionContext) {
+class LibraryService(repository: LibraryRepository, log: LoggingAdapter)(implicit ec: ExecutionContext) {
 
   def addNew(entry: LibraryEntry): Future[Int] =
-    db.run(repository.add(LibraryEntryDbDto.fromDomain(entry)))
+    repository.add(LibraryEntryDbDto.fromDomain(entry))
 
   def getRecords(offset: Option[Int] = None, limit: Option[Int] = None, statusFilter: Option[Status] = None): Future[Seq[LibraryEntry]] =
     if (offset.exists(_ < 0) || limit.exists(_ < 0))
       Future.failed(new IllegalArgumentException)
     else
       for {
-        entriesDto <- db.run(repository.list(statusFilter, offset, limit))
+        entriesDto <- repository.list(statusFilter, offset, limit)
         entriesFSeq = entriesDto.map(e => Future.fromTry(e.toDomain))
         entries <- Future.sequence(entriesFSeq)
       } yield entries
 
-
-  def changeStatus(uid: Uid, newStatus: Status): Future[Unit] =
-    db.run(repository.updateStatus(uid, newStatus)) flatMap {
-      case 0 => Future.failed(UidNotExists(s"There is no Library Entry for uid: $uid"))
-      case 1 => Future.successful(())
-    }
+  def changeStatus(uid: Uid, newStatus: Status): Future[LibraryEntry] = {
+    val uidNotExistsException = Future.failed(new IllegalArgumentException(s"Library Entry does not exist for uid: ${uid.value}"))
+
+    for {
+      entry       <-  repository.get(uid).flatMap {
+                        case Some(v) => Future.successful(v)
+                        case None => uidNotExistsException
```

Comment:
shouldn't that be done in repositories?

Reply:
My train of though was like that:
I want repository to return `Future[Option[LibraryEntryDbDto]]` since its representing the exact thing thats returned.
However in `LibraryService` I decide to handle this `Option` in some way to get `Future[LibraryEntryDbDto]`.
Possibly in some other part in the future this optionality would have to be handled differently, so I have chosen to do it in this way.
Also I do not believe future coming from repository should be a `Failure` when id doesnt exists - this is not an exceptional situation, which can be simply modelled with an `Option`.

Is this something you do think is not proper ?